Replace NEXT_PUBLIC_API_URI instead of API_URI

emreloper opened the pull request:
`PLUNK_API_URI` is a placeholder for `NEXT_PUBLIC_API_URI` inside the Dockerfile.

https://github.com/useplunk/plunk/blob/f39ff2a9886a796293f1a3fc0c386694af5015ce/Dockerfile#L8

`API_URI` variable could be an internal URI like `http://plunk:3000` while `NEXT_PUBLIC_API_URI` has to be public.

Separating internal and public URI variables can help with performance by avoiding network overhead for server requests.

This commit also solves the issue #114 by using an internal URI for `API_URI`.

driaug commented:
This would require quite some changing in documentation and templates.

I am also not sure if this would actually solve the #114 since the internal endpoint is probably not reachable if the public one isn't either. Adding retry logic to the CRON job might be the easier and more sustainable solution to this.

emreloper commented:
> This would require quite some changing in documentation and templates.
> 
> I am also not sure if this would actually solve the #114 since the internal endpoint is probably not reachable if the public one isn't either. Adding retry logic to the CRON job might be the easier and more sustainable solution to this.

I have a self-hosted instance on Hetzer with Cloudflare DNS. I was getting the same issue as #114. The cron job was not able to reach to the public URL after running a few times even if I could reach it on a browser. It could be related to Cloudflare security rules but I couldn't solve it.

After separating internal and public URLs for server/client requests, that issue was resolved. Right now the dashboard and sending transational mails work for me. I didn't have a new issue since 3 days.

However, I'm not familiar with the entire codebase. If this can cause issues, feel free to close the PR. I just wanted to help ✌️

emre-decrypt committed:
Replace NEXT_PUBLIC_API_URI instead of API_URI

`PLUNK_API_URI` is a placeholder for `NEXT_PUBLIC_API_URI` inside the Dockerfile. `API_URI` variable could be an internal URI like `http://plunk:3000` while `NEXT_PUBLIC_API_URI` has to be public.

Separating internal and public URI variables can help with performance by avoiding network overhead for server requests.

This commit also solves the issue #114.

driaug commented:
Closing this in preparation of `next` branch
